text/0400-appmonitor-l2.md

```diff
@@ -0,0 +1,499 @@
+declare const myRole: iam.IRole;
+
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: 'amazon.com',
+  appMonitorName: 'my-app-monitor',
+  authorizer: new rum.CognitoIdentityPoolAuthorizer({
+    identityPoolId: 'my-user-pool-id',
+    unauthenticatedRole: myRole,
+  }),
+});
+```
+
+#### Use Third-party provider
+
+You can also use a third party authenticator as in `CognitoIdentityPoolAuthorizer`.
+
+```ts
+import * as rum from '@aws-cdk/aws-rum';
+import * as iam from '@aws-cdk/aws-iam';
+
+declare const myRole: iam.IRole;
+
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: 'amazon.com',
+  appMonitorName: 'my-app-monitor',
+  authorizer: new rum.ThirdPartyAuthorizer({
+    role: myRole,
+  }),
+});
+```
+
+### Code Snippet
+
+AppMonitor generates a code snippet that looks like to create on the management console. Note, however,
+that unlike the management console, the code snippets do not have `<script>` tags because the CDK expects them to be automatically embedded in the application.
+
+```ts
+import * as rum from '@aws-cdk/aws-rum';
+
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: 'amazon.com',
+  appMonitorName: 'my-app-monitor'
+});
+
+// (function(n,i,v,r,s,c,x,z){...})
+const codeSnippet = appMonitor.generateCodeSnippet('CodeSnippet');
+```
+
+#### RUM web client configuration
+
+If you want to use [RUM web client arguments](https://github.com/aws-observability/aws-rum-web/blob/main/docs/cdn_installation.md#arguments)
+ (e.g applicationVersion, cookieAttributes), you can pass options to `generateCodeSnippet` argument.
+
+```ts
+import * as rum from '@aws-cdk/aws-rum';
+
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: 'amazon.com',
+  appMonitorName: 'my-app-monitor'
+});
+
+const codeSnippet = appMonitor.generateCodeSnippet('CodeSnippet', {
+  applicationVersion: '1.1.0',
+  configuration: {
+    pageIdFormat: rum.PageIdFormat.HASH
+  }
+});
+```
+
+#### With s3 deployment
+
+By using S3 Deployment, you can automate embeded to your application.
+This example is shortest way that deploy site using RUM.
+
+```ts
+import * as rum from '@aws-cdk/aws-rum';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as s3deploy from '@aws-cdk/aws-s3-deployment';
+
+const webSiteBucket = new s3.Bucket(this, 'WebSiteBucket', {
+  publicReadAccess: true,
+  websiteIndexDocument: 'index.html'
+});
+const appMonitor = new rum.AppMonitor(this, 'AppMonitor', {
+  domain: webSiteBucket.bucketWebsiteDomainName,
+  appMonitorName: 'my-app-monitor'
+});
+const codeSnippet = appMonitor.generateCodeSnippet('CodeSnippet');
+const rumJs = s3deploy.Source.data('rum.js', codeSnippet);
+
+const html = s3deploy.Source.data('index.html', `<html>
+  <head>
+    <script src="/rum.js" async="true"></script>
+  </head>
+  <body>Hello RUM</body>
+</html>`);
+
+new s3deploy.BucketDeployment(this, 'BucketDeployment', {
+  sources: [html, rumJs],
+  destinationBucket: webSiteBucket
+});
```

Comment:
Practically speaking, I don't think developers are going to write raw HTML inline like this. Also part of this is boilerplate, like specifying the script name `rum.js` twice. Can we make this simpler? And closer to a real-world scenario?